### ISSUE TITLE
Error trying to open a long conversation

### ISSUE THREAD

**Issue by Pauligrinder** [Foundry376/Mailspring#773]
Operating system: Arch Linux, x86_64
Mailspring version: 1.1.5-1 (from AUR, installed using apacman)

Mailspring started to get very sluggish when opening a long conversation (around 25 replies, four recipients, attachments and links in pretty much every reply, and so on), and finally after clicking the reply-all button it didn't open the conversation, but showed this error instead:

```
Error: Failed to execute 'setStart' on 'Range': The offset 4294967295 is larger than the node's length (82).
    at findDOMRange (/usr/share/mailspring/resources/app.asar/node_modules/slate-react/lib/utils/find-dom-range.js:37:5)
    at Content._this.updateSelection (/usr/share/mailspring/resources/app.asar/node_modules/slate-react/lib/components/content.js:154:46)
    at Content._this.componentDidMount (/usr/share/mailspring/resources/app.asar/node_modules/slate-react/lib/components/content.js:113:13)
    at commitLifeCycles (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:149:193)
    at b (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:158:229)
```

Closing and re-opening mailspring seems to have fixed it though.

**Comment by bengotow** [comment]
Hey! Thanks for reporting this—that's definitely no good. I'll see if I can reproduce this—it sounds like there was probably something specific about that conversation that caused this error.

**Comment by boostpaal** [comment]
Same thing happened to me, and now the preview window is not working (shows the error message independent of which email I choose).

**Comment by MarcBurgs** [comment]
Had the same issue. Fixed it by turning the reading pane off and on again in the View settings.

**Comment by Pauligrinder** [comment]
This happened again today, with the latest version, but this time as I was writing a reply to a shorter conversation.

```
mailspring -v
1.2.1-7e7447b6
```

```
Error: Failed to execute 'setStart' on 'Range': The offset 4294967294 is larger than the node's length (1).
    at findDOMRange (/usr/share/mailspring/resources/app.asar/node_modules/slate-react/lib/utils/find-dom-range.js:37:5)
    at Content._this.updateSelection (/usr/share/mailspring/resources/app.asar/node_modules/slate-react/lib/components/content.js:154:46)
    at Content._this.componentDidUpdate (/usr/share/mailspring/resources/app.asar/node_modules/slate-react/lib/components/content.js:121:13)
    at commitLifeCycles (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:149:308)
    at b (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:158:229)
    at m (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:169:217)
    at w (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:168:413)
    at batchedUpdates (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:171:357)
    at tc (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:54:269)
    at vd (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:71:249)
```

**Comment by Dekadinious** [comment]
I got this today after accidentally hitting the up-arrow too many times in a conversation. This is on windows with the newest version that got installed today. None of the suggestions above is working. I can't read any e-mail now.

`Error: Failed to execute 'setStart' on 'Range': The offset 4294967294 is larger than the node's length (1).
    at findDOMRange (C:\Users\tor\AppData\Local\Mailspring\app-1.2.2\resources\app.asar\node_modules\slate-react\lib\utils\find-dom-range.js:37:5)
    at Content._this.updateSelection (C:\Users\tor\AppData\Local\Mailspring\app-1.2.2\resources\app.asar\node_modules\slate-react\lib\components\content.js:154:46)
    at Content._this.componentDidMount (C:\Users\tor\AppData\Local\Mailspring\app-1.2.2\resources\app.asar\node_modules\slate-react\lib\components\content.js:113:13)
    at commitLifeCycles (C:\Users\tor\AppData\Local\Mailspring\app-1.2.2\resources\app.asar\node_modules\react-dom\cjs\react-dom.production.min.js:149:193)
    at b (C:\Users\tor\AppData\Local\Mailspring\app-1.2.2\resources\app.asar\node_modules\react-dom\cjs\react-dom.production.min.js:158:229)
    at m (C:\Users\tor\AppData\Local\Mailspring\app-1.2.2\resources\app.asar\node_modules\react-dom\cjs\react-dom.production.min.js:169:217)
    at w (C:\Users\tor\AppData\Local\Mailspring\app-1.2.2\resources\app.asar\node_modules\react-dom\cjs\react-dom.production.min.js:168:413)
    at z (C:\Users\tor\AppData\Local\Mailspring\app-1.2.2\resources\app.asar\node_modules\react-dom\cjs\react-dom.production.min.js:167:32)
    at u (C:\Users\tor\AppData\Local\Mailspring\app-1.2.2\resources\app.asar\node_modules\react-dom\cjs\react-dom.production.min.js:165:441)
    at Object.enqueueSetState (C:\Users\tor\AppData\Local\Mailspring\app-1.2.2\resources\app.asar\node_modules\react-dom\cjs\react-dom.production.min.js:108:299)
    at _class.A.setState (C:\Users\tor\AppData\Local\Mailspring\app-1.2.2\resources\app.asar\node_modules\react\cjs\react.production.min.js:12:357)
    at _draftStore2.default.sessionForClientId.then.session (/src\decorators\inflates-draft-client-id.jsx:66:14)
    at <anonymous>`

**Comment by ajdonnison** [comment]
Happened to me with up arrow in replying to a thread of 6 messages.  Had to quit and restart. On Linux:
```
Error: Failed to execute 'setStart' on 'Range': The offset 4294967294 is larger than the node's length (1).
    at findDOMRange (/snap/mailspring/216/usr/share/mailspring/resources/app.asar/node_modules/slate-react/lib/utils/find-dom-range.js:37:5)
    at Content._this.updateSelection (/snap/mailspring/216/usr/share/mailspring/resources/app.asar/node_modules/slate-react/lib/components/content.js:154:46)
    at Content._this.componentDidUpdate (/snap/mailspring/216/usr/share/mailspring/resources/app.asar/node_modules/slate-react/lib/components/content.js:121:13)
    at commitLifeCycles (/snap/mailspring/216/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:149:308)
    at b (/snap/mailspring/216/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:158:229)
    at m (/snap/mailspring/216/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:169:217)
    at w (/snap/mailspring/216/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:168:413)
    at batchedUpdates (/snap/mailspring/216/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:171:357)
    at tc (/snap/mailspring/216/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:54:269)
    at vd (/snap/mailspring/216/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production
```

**Comment by bengotow** [comment]
Hey folks! This should be addressed in the latest release (1.6.3) that shipped earlier this week. The app should auto-update on it's own, or you can grab the new version from the releases page: https://github.com/Foundry376/Mailspring/releases/. Thanks for filing this - enjoy the update!
